src/plugins/workspace/public/components/delete_workspace_modal/delete_workspace_modal.tsx

```diff
@@ -31,9 +50,119 @@ export function DeleteWorkspaceModal(props: DeleteWorkspaceModalProps) {
   const [value, setValue] = useState('');
   const { onClose, selectedWorkspace, returnToHome } = props;
   const {
-    services: { application, notifications, http, workspaceClient },
+    services: { application, notifications, http, workspaceClient, savedObjects, workspaces },
   } = useOpenSearchDashboards<{ workspaceClient: WorkspaceClient }>();
 
+  const savedObjectsClient = savedObjects!.client;
+  const [workspaceOptions, setWorkspaceOptions] = useState<WorkspaceOption[]>([]);
+  const [targetWorkspaceOption, setTargetWorkspaceOption] = useState<WorkspaceOption[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const targetWorkspaceId = targetWorkspaceOption?.at(0)?.key;
+  let confirmDuplicateButtonEnabled = false;
+  const [savedObjectsCount, setSavedObjectsCount] = useState(0);
+  const [allowedTypes, setAllowedTypes] = useState<string[]>([]);
+  const maxObjectsAmount: number = 200;
+  const onTargetWorkspaceChange = (targetOption: WorkspaceOption[]) => {
+    setTargetWorkspaceOption(targetOption);
+  };
+
+  if (!!targetWorkspaceId && savedObjectsCount > 0) {
+    confirmDuplicateButtonEnabled = true;
+  }
+
+  useEffect(() => {
+    const workspaceList = workspaces!.workspaceList$.value;
```

Comment:
There is a `useObservable` hook, you should get workspaceList like `const workspaceList = useObservable(worksapces.workspaceList$)` out of this useEffect and makes workspaceList a dependency of this useEffect.

Reply:
I will use the useObservable hook.